### PR TITLE
AP-1881 Add debugging and reset ccms_reason

### PR DESCRIPTION
**Pull request by stephenrichards**
## Add debugging and reset ccms_reason

[Link to story](https://dsdmoj.atlassian.net/browse/AP-1881)

This PR fixes two issues, and introduces debugging to track down another

Fixed: 
 - the ccms_reason wasn't being reset to nil in all cases when transitioning from use_ccms to another state.
 - adds `expire_now` to the `Admin::ReportsController` actions to produce the CSV files to prevent caching and responding with stale data.

Debugging:
 - in all cases where we transition to use_ccms state with a reason of unknown, print a back trace and the referer URL

## Checklist

Before you ask people to review this PR:

- [ ] Tests and rubocop should be passing: `bundle exec rake`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [ ] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [ ] The PR description should say what you changed and why, with a link to the JIRA story.
- [ ] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/models/concerns/base_state_machine.rb [hunk 1]
```diff
@@ -83,6 +83,9 @@ class BaseStateMachine < ApplicationRecord  # rubocop:disable Metrics/ClassLengt
                   after: proc { |reason|
                     raise 'Invalid ccms_reason' unless reason.in?(VALID_CCMS_REASONS)
 
+                    # logging to find out where unknonw ccms reasons are coming from
+                    Raven.capture_message "UNKNOWN CCMS REASON\n#{caller}" if reason.to_s == 'unknown'
```
Comment (Alex-Swann) [inline app/models/concerns/base_state_machine.rb:87]: couldn't you just do `reason == :unknown` to save having to create strings?

**Review by Alex-Swann — APPROVED** [review]
LGTM bar one minor comment

File: app/models/concerns/base_state_machine.rb (hunk 1 above)
Comment (stephenrichards) [reply app/models/concerns/base_state_machine.rb:87]: Well, I don't know if the reason is being passed in as a symbol or a string, so this makes sure it captures both possibilities

Comment (Alex-Swann) [reply app/models/concerns/base_state_machine.rb:87]: `reason.to_sym == :unknown` then?

**Review by Alex-Swann — APPROVED** [review]
Thanks for the explanation. It makes it clear what the additional bit you've added is doing.

**Pull request merged**